src/emmental/model.py

```diff
@@ -526,21 +532,23 @@ def predict(
                 if return_action_outputs and out_bdict:
                     for task_name in out_bdict.keys():
                         for action_name in out_bdict[task_name].keys():
-                            out_dict[task_name][action_name].extend(
-                                out_bdict[task_name][action_name]
-                            )
+                            if out_dict[task_name][action_name] == []:
```

Comment:
Can we move this to `merge_objects`?

Reply:
problem is that `out_dict[task_name][action_name]` is always going to be a list, but out_bdict[task_name][action_name] can be any type. Thus, the `merge_objects` function will just raise a type error because of type mistmatch

Reply:
this should be gone right?